header/header_modifier.go

```diff
@@ -29,31 +29,42 @@ func init() {
 
 type modifier struct {
 	name, value string
+	add         bool
```

Comment:
I'm trying to think of a better way to name / represent this...

We have 2 "modes" - Append or Replace - so maybe instead of the `add bool`, we could have a Mode that's an int that represents the modes that are defined with something like:

```
const (
  ReplaceValues = iota
  AppendValues
)
```
and then the struct could be:
```
type modifier struct {
  name, value string
  behavior    int
}

```

Reply:
What would that look like on the wire?

```
{
  "name": ...
 ...
  "behavior": "replace" | "append"
}
```
?

Another option might be to still have a boolean flag but invert it and default it to true

```
{
  "name": ...
  ...
  "replace": false
}
```

Reply:
Yeah, I was thinking there would be a "behavior" key that was either "replace" or "append"

Your modifierJSON struct would have a 
```
Behavior string `json:"behavior"`
```

and then in modifierFromJSON you'd have a check like
```
if msg.Behavior == "append" {
  modifier.SetBehavior(AppendValues)
}
```